scipy/signal/lfilter.c.src

```diff
@@ -94,34 +95,40 @@ convert_shape_to_errmsg(npy_intp ndim, npy_intp *Xshape, npy_intp *Vishape,
         expect_size = j != theaxis ? Xshape[j] : val;
 
         if (j == ndim - 1) {
-            tmp = PyUString_FromFormat("%" NPY_INTP_FMT, expect_size);
-            tmp1 = PyUString_FromFormat("%" NPY_INTP_FMT, Vishape[j]);
+            tmp = PyUnicode_FromFormat("%" NPY_INTP_FMT, expect_size);
+            tmp1 = PyUnicode_FromFormat("%" NPY_INTP_FMT, Vishape[j]);
         } else {
-            tmp = PyUString_FromFormat("%" NPY_INTP_FMT ",", expect_size);
-            tmp1 = PyUString_FromFormat("%" NPY_INTP_FMT ",", Vishape[j]);
+            tmp = PyUnicode_FromFormat("%" NPY_INTP_FMT ",", expect_size);
+            tmp1 = PyUnicode_FromFormat("%" NPY_INTP_FMT ",", Vishape[j]);
         }
         if (!tmp) {
             Py_DECREF(msg);
             Py_DECREF(msg1);
             Py_XDECREF(tmp1);
             return 0;
         }
+        /* Should this be merged with above? */
         if (!tmp1) {
             Py_DECREF(msg);
             Py_DECREF(msg1);
             Py_DECREF(tmp);
             return 0;
         }
```

Comment:
I can't understand how tmp would be true but tmp1 wouldn't be.

Reply:
Not sure, but this is more robust and I don't see a downside to leaving it as is.

Reply:
could you remove the comment above the block again?

Reply:
`PyUnicode_FromFormat` can fail, in which case `tmp` or `tmp1` may be null.  You could merge the two conditions using `Py_XDECREF` which allows null.